client/cypress/integration/dashboard/parameter_spec.js

```diff
@@ -0,0 +1,161 @@
+      .contains("Edit Parameters")
+      .click();
+  };
+
+  const saveMappingOptions = (close = true) => {
+    cy.getByTestId("EditParamMappingPopover")
+      .filter(":visible")
+      .within(() => {
+        cy.contains("button", "OK").click();
+      });
+
+    if (close) {
+      cy.contains("button", "OK").click();
+    }
+  };
+
+  const setWidgetParametersToDashboard = parameters => {
+    each(parameters, ({ name: paramName }, i) => {
+      cy.getByTestId(`EditParamMappingButton-${paramName}`).click();
+      cy.getByTestId("NewDashboardParameterOption")
+        .filter(":visible")
+        .click();
+      saveMappingOptions(i === parameters.length - 1);
+    });
+  };
+
+  it("supports widget parameters", function() {
+    // widget parameter mapping is the default for the API
+    cy.getByTestId(this.widgetTestId).within(() => {
+      cy.getByTestId("TableVisualization").should("contain", "example1");
+
+      cy.getByTestId("ParameterName-param1")
+        .find("input")
+        .type("{selectall}Redash");
+
+      cy.getByTestId("ParameterApplyButton").click();
+
+      cy.getByTestId("TableVisualization").should("contain", "Redash");
+    });
+
+    cy.getByTestId("DashboardParameters").should("not.exist");
+  });
+
+  it("supports dashboard parameters", function() {
+    openMappingOptions(this.widgetTestId);
+    setWidgetParametersToDashboard(parameters);
+
+    cy.getByTestId(this.widgetTestId).within(() => {
+      cy.getByTestId("ParameterName-param1").should("not.exist");
+    });
+
+    cy.getByTestId("DashboardParameters").within(() => {
+      cy.getByTestId("ParameterName-param1")
+        .find("input")
+        .type("{selectall}DashboardParam");
+
+      cy.getByTestId("ParameterApplyButton").click();
+    });
+
+    cy.getByTestId(this.widgetTestId).within(() => {
+      cy.getByTestId("TableVisualization").should("contain", "DashboardParam");
+    });
+  });
+
+  it("supports static values for parameters", function() {
+    openMappingOptions(this.widgetTestId);
+    cy.getByTestId("EditParamMappingButton-param1").click();
+
+    cy.getByTestId("StaticValueOption").click();
+
+    cy.getByTestId("EditParamMappingPopover").within(() => {
+      cy.getByTestId("ParameterValueInput")
+        .find("input")
+        .type("{selectall}StaticValue");
+    });
+
+    saveMappingOptions(true);
+
+    cy.getByTestId(this.widgetTestId).within(() => {
+      cy.getByTestId("ParameterName-param1").should("not.exist");
+    });
+
+    cy.getByTestId("DashboardParameters").should("not.exist");
+
+    cy.getByTestId(this.widgetTestId).within(() => {
+      cy.getByTestId("TableVisualization").should("contain", "StaticValue");
+    });
+  });
+
+  it("reorders parameters", function() {
+    // Reorder is only available on edit mode
+    cy.getByTestId("DashboardMoreButton").click();
+    cy.getByTestId("DashboardMoreButtonMenu")
+      .contains("li", "Edit")
+      .click();
```

Comment:
```suggestion
    // Reorder is only available on edit mode
    editDashboard();
```

importing from
https://github.com/getredash/redash/blob/de052ff02be9ff8d4187d273364c78ca22d65eb0/client/cypress/support/dashboard/index.js#L21-L27